### PR TITLE
fix(cozy-lib,harbor,mariadb): digest-pin cleanup-hook kubectl image and route it through cozy-lib.images-registry

### LINKED ISSUE #3088: fix(images): digest-pin cleanup-hook kubectl image and route it through cozy-lib.images-registry (harbor, mariadb)
## Describe the bug

The post-delete cleanup Jobs in several app charts hardcode the cleanup image as a
moving Docker Hub tag, bypassing both digest pinning and the Cozystack cluster
images registry:

- `packages/apps/harbor/templates/hooks/cleanup.yaml:30` — `image: docker.io/clastix/kubectl:v1.32`
- `packages/apps/mariadb/templates/hooks/cleanup-pvc.yaml:23` — `image: docker.io/clastix/kubectl:v1.32`

This was raised during review of #3077 (Harbor delete-time cleanup). It is **not a
regression introduced by that PR** — the same pattern already exists in the MariaDB
cleanup hook — but #3077 duplicates it onto a second chart, so it is worth fixing
both at once before the pattern fans out further.

### Two distinct problems with this line

1. **Moving tag, not pinned by digest.** `v1.32` is mutable: the upstream
   `clastix/kubectl` repository can re-push the tag to a different image at any
   time. This breaks build reproducibility (same git SHA pulls a different binary
   over time) and is a supply-chain risk (a compromised upstream tag silently
   ships into the cluster).

2. **Not routed through `cozy-lib.images-registry`.** Cozystack exposes a
   cluster-wide registry helper for air-gapped / mirrored installs:

   ```
   packages/library/cozy-lib/templates/_cozyconfig.tpl:26
   {{- define "cozy-lib.images-registry" -}}
   {{- (index .Values._cluster "images-registry") | default "" }}
   {{- end }}
   ```

   Because the image hardcodes `docker.io`, an air-gapped cluster (no Docker Hub
   egress, everything mirrored internally) cannot pull it — and it fails at the
   worst possible moment: **during app uninstall**, when the `post-delete` hook
   runs and the operator has the least ability to intervene. A stuck/failed
   cleanup Job blocks teardown of the release.

## Expected behaviour

The cleanup image should be:

- **digest-pinned**, e.g. `clastix/kubectl:v1.32@sha256:<digest>` (tag kept for
  readability, but resolution happens by immutable digest);
- **routed through the cluster images registry**, e.g.

  ```yaml
  image: {{ include "cozy-lib.images-registry" . }}/clastix/kubectl:v1.32@sha256:<digest>
  ```

  so mirrored / air-gapped installs resolve the image from the internal registry.

## Actual behaviour

Both cleanup Jobs reference `docker.io/clastix/kubectl:v1.32` directly. On a
standard cluster this works; on an air-gapped/mirrored cluster the `post-delete`
hook Pod fails to pull the image and the cleanup never runs (orphaned PVCs / ACME
artifacts remain, and depending on hook semantics the uninstall may stall).

## Scope / acceptance criteria

- [ ] `packages/apps/harbor/templates/hooks/cleanup.yaml` image is digest-pinned and routed through `cozy-lib.images-registry`
- [ ] `packages/apps/mariadb/templates/hooks/cleanup-pvc.yaml` image is digest-pinned and routed through `cozy-lib.images-registry`
- [ ] Helm unit tests updated/added to assert the rendered image goes through the registry helper
- [ ] Confirm the chosen pattern is correct when `images-registry` is empty (default `""`) — the image must still resolve on a standard install (no leading `/`)

## Notes

- `cozy-lib.images-registry` is currently only **defined**, not consumed by any
  chart template (`grep` finds only its definition). This issue is therefore also
  the opportunity to establish the canonical "how to reference a cluster image"
  pattern, including the empty-default edge case above.
- A dedicated digest needs to be resolved for `clastix/kubectl:v1.32` and recorded
  in the chart(s).

## Additional context

Raised in PR #3077 review by @IvanHunters (`[MINOR]` finding): "Cleanup image
`docker.io/clastix/kubectl:v1.32` is a moving tag, not pinned by digest, and is
not routed through `cozy-lib.images-registry` ... Recommend digest-pinning and
routing through `cozy-lib.images-registry` in both before this is fanned out
further."

### PR DESCRIPTION
**Pull request by scooby87**
## What this PR does

Fixes #3088.

The post-delete cleanup Jobs in the **harbor** and **mariadb** charts hardcoded the cleanup image as `docker.io/clastix/kubectl:v1.32` — a moving Docker Hub tag. This:

- **broke reproducibility / was a supply-chain risk** — `v1.32` is mutable and can be re-pushed to a different image;
- **could not be satisfied on air-gapped / mirrored clusters** — the hardcoded `docker.io` bypassed the cluster images registry, and it failed at the worst moment: during **uninstall**, when the `post-delete` hook runs and a stuck Job stalls release teardown.

### Changes

- **New canonical helper `cozy-lib.image`** (`packages/library/cozy-lib/templates/_cozyconfig.tpl`). Given a registry-relative image path and the global context, it prefixes `cozy-lib.images-registry` when set (`<registry>/<image>`) and returns the image unchanged when empty — **with no leading `/`**, so standard installs still resolve from the default registry. This establishes the previously-undefined "how to reference a cluster image" pattern (`images-registry` was defined but never consumed).
- **Digest-pinned image** in both cleanup hooks: `clastix/kubectl:v1.32@sha256:b9ef7d8dbe65bcc81a46c09b8dc7543103055021c4f43287bf59e92a8f4fe05c` (the immutable multi-arch OCI index for the `v1.32` tag; the same digest already pinned for `postgres-operator`'s webhook hook).
- **Renovate keeps the pin fresh.** The repo disables the helm-values manager (`enabledManagers`), and the ref is assembled at render time by the helper rather than being a Dockerfile / go.mod dependency, so the built-in managers can't see it. A `custom.regex` manager over the hook templates tracks `clastix/kubectl` by tag+digest — mirroring the existing `extra/etcd` manager — so the supply-chain pin doesn't silently age.
- **Hardened `cozy-lib.images-registry`** against a nil `.Values._cluster` so the helper never crashes the render — relevant precisely because this hook renders during teardown.
- **Helm unit tests** in both charts assert the rendered image for three cases: empty registry (default), registry set, and the no-leading-slash edge case. Assertions match by **pattern** (routed + digest-pinned + no leading slash) rather than an exact digest, so a Renovate bump keeps them green.

### Verification

- `helm unittest` passes for both charts (harbor: 16, mariadb: 10).
- Real `helm template` renders confirmed for both charts:
  - empty registry → `clastix/kubectl:v1.32@sha256:b9ef…`
  - `images-registry=registry.internal:5000` → `registry.internal:5000/clastix/kubectl:v1.32@sha256:b9ef…`
- The Renovate `matchStrings` regex verified to capture `currentValue=v1.32` and the digest from both hook templates.

### Note on the wider footprint

The same `docker.io/clastix/kubectl:v1.32` hardcode is already merged on `main` in several other cleanup hooks (bucket, qdrant, gateway, seaweedfs, kubernetes, tenant, dashboard, keycloak-configure) and appears in the open PRs #3170 (etcd), #3094 (monitoring), #3072 (clickhouse). This PR fixes harbor + mariadb per the issue and establishes the `cozy-lib.image` + Renovate pattern; migrating the remaining hooks is a follow-up.

### Release note

```release-note
fix(cozy-lib,harbor,mariadb): digest-pin the harbor and mariadb post-delete cleanup-hook kubectl image and route it through the cluster images registry (`cozy-lib.images-registry`) via the new `cozy-lib.image` helper, so mirrored / air-gapped installs can resolve it and uninstall no longer depends on a moving Docker Hub tag; a custom Renovate manager keeps the digest fresh
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Updated post-delete cleanup hook Jobs to use digest-pinned container images for more secure, repeatable execution.
  * Improved image reference rendering so cleanup hooks work correctly with or without an image registry configured, avoiding invalid leading/trailing slash issues.
* **Tests**
  * Extended cleanup hook rendering tests to validate digest pinning and correct registry prefix behavior, including edge cases.
* **Chores**
  * Enhanced automated dependency detection to treat digest-pinned hook image references as stable.

### CODE REVIEW

File: packages/library/cozy-lib/templates/_cozyconfig.tpl [hunk 1]
```diff
@@ -24,7 +24,25 @@ Get the images registry.
...
+"<registry>/<image>"; when empty (the default on a standard install) the image
+is returned unchanged so it resolves from its default registry — importantly
+with no leading "/", which would otherwise be an invalid reference.
+*/}}
+{{- define "cozy-lib.image" -}}
+{{- include "cozy-lib.checkInput" . }}
+{{- $image := index . 0 }}
+{{- $ctx := index . 1 }}
+{{- $registry := include "cozy-lib.images-registry" $ctx }}
+{{- with $registry }}{{ . }}/{{ end }}{{ $image }}
```
[the remark this answers is not in the document]
Comment (scooby87) [reply packages/library/cozy-lib/templates/_cozyconfig.tpl]: Good catch — applied in a76c5929f. The `cozy-lib.image` helper now `trimPrefix "/"` on the image path and `trimSuffix "/"` on the registry, so a misconfigured `images-registry` with a trailing slash (e.g. `registry.example.com/`) no longer yields a double slash. Added a regression test in both the harbor and mariadb cleanup-hook suites asserting the rendered ref contains no `//`. Thanks!

**Review by IvanHunters — APPROVED** [review]
## Review: LGTM with non-blocking notes

The digest-pin and the `cozy-lib.image` routing are correct across all three charts. Both chart unittests pass (harbor 16, mariadb 10).

**Main note (accuracy of framing)**
- `cozy-lib.images-registry` reads `.Values._cluster["images-registry"]`, but that key is never populated anywhere in the tree. The `cozystack-values` emitter (`packages/core/platform/templates/apps.yaml`) emits ~40 `_cluster` keys and `images-registry` is not one of them; there is no Go injector and no installer manifest that sets it. So on every real cluster today the registry resolves empty and `cozy-lib.image` returns the image unchanged. The routing is correct and future-proof, but it is currently dormant: air-gapped resolution is not achieved until a separate change populates `_cluster.images-registry`. Suggest softening the PR body / release-note to "establishes the routing" rather than implying present air-gapped capability.

**Verified**
- `helm template` renders `clastix/kubectl:v1.32@sha256:b9ef7d8…` with empty registry (no leading slash) and `registry.example.com/clastix/kubectl:v1.32@sha256:b9ef7d8…` with the registry set.
- nil-`_cluster` hardening (`(.Values._cluster | default dict)`) is necessary: the old `index .Values._cluster "images-registry"` fails with `index of untyped nil` when `_cluster` is unset.
- Bare `clastix/kubectl` (no `docker.io/` prefix) resolves to `docker.io/clastix/kubectl`, so the join produces `<mirror>/clastix/kubectl` rather than `<mirror>/docker.io/clastix/kubectl`.
- Digest matches the pin already at `packages/system/postgres-operator/values.yaml`.
- Renovate manager `matchStrings` requires `@sha256:` and `managerFilePatterns` matches both changed files.

**Follow-ups (separate PR, as the body acknowledges)**
- Migrate the remaining hardcoded `docker.io/clastix/kubectl:v1.32` cleanup hooks to the same pattern (keycloak-configure, dashboard, bucket, qdrant, kubernetes, seaweedfs, gateway, clickhouse, tenant).
- Some of those are not under a `templates/hooks/` directory, so they are not caught by this PR's Renovate `managerFilePatterns`; broaden the pattern when migrating them or the pins will silently age.

Upgrade impact: none (hooks fire only on post-delete; rendered image is registry-identical to before plus an immutable digest). Fresh install: none negative.
